spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java

```diff
@@ -787,6 +790,39 @@ private void addFormatters(FormatterRegistry registry) {
 		}
 	}
 
+	/**
+	 * Parse a list of partitions into a {@link List}. Example: "0-5,10-15".
+	 * @param partsString the comma-delimited list of partitions/ranges.
+	 * @return the partitions sorted and de-duplicated.
+	 * @since 2.6.4
+	 */
+	private List<Integer> parsePartitions(String partsString) {
+		String[] partsStrings = partsString.split(",");
+		if (partsStrings.length == 1 && !partsStrings[0].contains("-")) {
+			return Collections.singletonList(Integer.parseInt(partsStrings[0].trim()));
+		}
+		List<Integer> parts = new ArrayList<>();
+		for (String part : partsStrings) {
+			if (part.contains("-")) {
+				String[] startEnd = part.split("-");
+				Assert.state(startEnd.length == 2, "Only one hyphen allowed for a range of partitions: " + part);
+				int start = Integer.parseInt(startEnd[0].trim());
+				int end = Integer.parseInt(startEnd[1].trim());
+				Assert.state(end >= start, "Invalid range: " + part);
+				for (int i = start; i <= end; i++) {
+					parts.add(i);
+				}
+			}
+			else {
+				parsePartitions(part).stream().forEach(p -> parts.add(p));
+			}
+		}
+		return parts.stream()
+				.sorted()
+				.distinct()
+				.collect(Collectors.toList());
```

Comment:
Looks like you do a double `collect()`.
This method is `private` and if you don't want to mix concern with that `map()` in here, I think it is fully safe just return a `Stream<Integer>`  for the further processing rather than re-stream the list one more time.